Core/Tornado/Server/TornadoServer.py

```diff
@@ -0,0 +1,214 @@
+
+    :param list services: List of services you want to start, start all by default
+    :param str debug: Activate debug mode of Tornado (autoreload server + more errors display) and M2Crypto
+    :param int port: Used to change port, default is 443
+    """
+
+    # If port not precised, get it from config
+    if port is None:
+      port = gConfig.getValue("/Systems/Tornado/%s/Port" % PathFinder.getSystemInstance('Tornado'), 443)
+
+    if services and not isinstance(services, list):
+      services = [services]
+
+    # URLs for services: 1URL/Service
+    self.urls = []
+    # Other infos
+    self.debugSSL = debugSSL  # Used by tornado and M2Crypto
+    self.port = port
+    self.handlerManager = HandlerManager()
+    self._monitor = MonitoringClient()
+    self.__monitoringLoopDelay = 60  # In secs
+
+    # If services are defined, load only these ones (useful for debug purpose or specific services)
+    if services:
+      retVal = self.handlerManager.loadHandlersByServiceName(services)
+      if not retVal['OK']:
+        gLogger.error(retVal['Message'])
+        raise ImportError("Some services can't be loaded, check the service names and configuration.")
+
+    # if no service list is given, load services from configuration
+    handlerDict = self.handlerManager.getHandlersDict()
+    for item in handlerDict.iteritems():
+      # handlerDict[key].initializeService(key)
+      self.urls.append(url(item[0], item[1], dict(debug=self.debugSSL)))
+    # If there is no services loaded:
+    if not self.urls:
+      raise ImportError("There is no services loaded, please check your configuration")
+    self.__report = None
+    self.__monitorLastStatsUpdate = None
+
+  def startTornado(self, multiprocess=False):
+    """
+      Start the tornado server when ready.
+      The script is blocked in the Tornado IOLoop.
+      Multiprocess option is available but may be used with caution
+    """
+
+    gLogger.debug("Starting Tornado")
+    self._initMonitoring()
+
+    if self.debugSSL:
+      gLogger.warn("Server is running in debug mode")
+
+    router = Application(self.urls, debug=self.debugSSL)
+
+    certs = Locations.getHostCertificateAndKeyLocation()
+    if certs is False:
+      gLogger.fatal("Host certificates not found ! Can't start the Server")
+      raise ImportError("Unable to load certificates")
+    ca = Locations.getCAsLocation()
+    ssl_options = {
+        'certfile': certs[0],
+        'keyfile': certs[1],
+        'cert_reqs': M2Crypto.SSL.verify_peer,
+        'ca_certs': ca,
+        'sslDebug': self.debugSSL
+    }
+
+    self.__monitorLastStatsUpdate = time.time()
+    self.__report = self.__startReportToMonitoringLoop()
+
+    # Starting monitoring, IOLoop waiting time in ms, __monitoringLoopDelay is defined in seconds
+    tornado.ioloop.PeriodicCallback(self.__reportToMonitoring, self.__monitoringLoopDelay * 1000).start()
+
+    # Start server
+    server = HTTPServer(router, ssl_options=ssl_options)
```

Comment:
Correct. again, this work is the initial work by my student, refreshed and fixed. At the time, I considered the compression to be a further optimization to be done. It can probably be added to that PR though.

Reply:
Added

Reply:
added: responses are compressed, requests CAN be decompressed, but are not compressed because it requires manual work with `requests`, and probably shows no advantages for normal uses cases.